docs/multiple_image_support_design_proposal.md

```diff
@@ -0,0 +1,549 @@
+                                              * be available to satisfy
+                                              * compliance
+                                              */
+};
+```
+
+*Note: IETF manifest proposal suggest to use exact hash value instead of version
+number to describe dependency*
+
+In the manifest section can be more then one dependency entry, but in practice
+if the platform only supports two individual images then there can be maximum
+one entry which reflects to the other image. I.e.: Non-secure image can be
+dependent on secure image and vice versa.
+
+Dependency is a mutual relation. If one image depends on an other then that
+other image also depend on the first one. Therefore in case of dependency there
+must be one dependency entry in both images manifest in order to ensure proper
+updates.
+
+
+Table about describing dependency:
+
+|Image 0 version|Comment|Dependency entry||Image 1 version|Comment|Dependency entry|
+|---------------|-------|---------------|-------|
+|1.0.0.0|First release|2.0.0.0||2.0.0.0|Second release|1.0.0.0|
+|1.0.0.1|Bug fix|2.0.0.0||2.0.0.0|no change|1.0.0.0|
+|1.0.0.2|Bug fix|2.0.0.0||2.0.0.0|no change|1.0.0.0|
+|1.1.0.0|API change|2.1.0.0||2.1.0.0|API change|1.1.0.0|
+|1.1.0.0|no change|2.1.0.0||2.2.0.0|Security patch|1.1.0.0|
+|1.1.0.1|Bug fix|2.2.0.0||2.2.0.0|Security patch|1.1.0.0|
+
+`Note: Might be this double linked dependency handling can be replaced with
+something simpler, i.e.: single linked. That case should be a main image, which
+has no dependency and other images can depend only on previous image in the
+chain.`
+
+# How to include dependency entries?
+In case of a single vendor, who owns all the images but still want to do
+independent updates for example to save power, to including dependency entries
+as a TLV is pretty easy. Only the scripts should be modified.
+
+In case of multi vendor scenario when the device maker/owner only owns the main
+application and other SW images is provided by other party, then it might, not
+straight forward to ask the other party to include my main application's version
+to his image as a TLV entry. Might be TLV entries must be extended by device
+owner and that part should be signed his own key?
+
+# Integrity protection of manifest
+Currently the image manifest is not needed to be part of the integrity protected
+section. There is no security vulnerability that could be exploited by modifying
+it. It contains these TLV entries:
+  - Image hash
+  - Key hash
+  - Signature (hash based)
+
+```
+  +--------------------+
+  | Header             |   <- struct image_header
+  +--------------------+
+  | Payload            |
+  +--------------------+
+  | Manifest           |
+  | +----------------+ |
+  | | Manifest header| |   <- struct image_tlv_info
+A | +----------------+ |
+  | | SHA256 hash    | |   <- hash of header and payload (struct image_tlv)
+B | +----------------+ |
```

Comment:
Thinking about this a bit more, I think we should make the boundary of 'B' more explicit in the manifest header. It should be very clear which part is covered by the signature, and which part is the signature. It is a fairly minor change, basically putting two size fields in the manifest header. The idea is that on start, the code would verify the signature, and then after that rely on all of the previous manifest information being valid (at least until reboot).

Reply:
Good idea. Are the two size field pointing to the borders of the signature protected area?